src/gui/shell.cpp

```diff
@@ -426,14 +429,14 @@ void Shell::setNeovimCursor(quint64 row, quint64 col)
 	update(QRect(neovimCursorTopLeft(), neovimCharSize()));
 }
 
-void Shell::handleNormalMode(QPainter& painter)
+void Shell::handleModeChange(const QString& mode)
 {
-	m_insertMode = false;
-}
-
-void Shell::handleInsertMode(QPainter& painter)
-{
-	m_insertMode = true;
+	// TODO: Implement visual aids for other modes
+	if (mode == "insert") {
```

Comment:
Not yet, I've just changed the minimum amount to keep showing the thin cursor in insert mode. For replace mode (at least in gvim) the mode cursor looks different (like a thick underline).